antarest/login/repository.py

```diff
@@ -1,36 +1,62 @@
 import uuid
-from typing import Dict, Optional, List
+from contextlib import contextmanager
+from typing import Dict, Optional, List, Generator, Any
+
+from sqlalchemy.engine import Engine  # type: ignore
+from sqlalchemy.orm import Session, sessionmaker  # type: ignore
 
 from antarest.common.config import Config
-from antarest.login.model import User, Role
+from antarest.login.model import User, Role, Password
+
+
+@contextmanager
+def session_scope(engine: Engine) -> Generator[Session, Any, Any]:
+    """Provide a transactional scope around a series of operations."""
+    try:
+        Session = sessionmaker(engine, expire_on_commit=False)
+        sess = Session()
+        yield sess
+        sess.commit()
+    except:
+        sess.rollback()
+        raise
+    finally:
+        sess.close()
 
 
 class UserRepository:
-    def __init__(self, config: Config) -> None:
-        self.users: Dict[int, User] = {
-            0: User(
-                id=0,
+    def __init__(self, config: Config, engine: Engine) -> None:
```

Comment:
It should be a request-scoped session indeed. Today, we have choice between server-scoped session or atomic-scoped session. To avoid endlessly session, I code a atomic-scoped session.